File: crates/offprint/examples/capture_memory.rs
use futures_util::StreamExt as _;
use offprint::{CaptureOutput, Offprint};

#[tokio::main]
async fn main() -> offprint::Result<()> {
    let offprint = Offprint::new()?;
    let job = offprint
        .capture("https://example.com")?
        .output(CaptureOutput::memory(16 * 1024 * 1024))
        .start()
        .await?;
    let mut events = job.events();
    while let Some(event) = events.next().await {
        if event.is_terminal() {
            break;
        }
    }

    let receipt = job.result().await?;
    assert!(receipt.resources.is_complete());
    offprint.close().await?;
    Ok(())
}

File: sdk/python/examples/capture_memory.py
import asyncio

from offprint import Offprint


async def capture_memory() -> None:
    async with Offprint() as offprint:
        request = offprint.captures.request("https://example.com")
        request["output"] = {"kind": "memory", "maxBytes": 16 * 1024 * 1024}
        job = await offprint.captures.start(request)
        async for event in job.events():
            if event["type"] == "warning":
                print(event["warning"]["code"])

        receipt = await job.result()
        artifact = receipt["artifact"]
        if artifact["kind"] != "bytes":
            raise RuntimeError("expected a memory artifact")
        html = bytes(artifact["content"])
        print(len(html), artifact["sha256"])


asyncio.run(capture_memory())
